programs/ssl/ssl_client2.c

```diff
@@ -116,7 +116,7 @@ int main( void )
 #define DFL_EXTENDED_MS         -1
 #define DFL_ETM                 -1
 
-#define GET_REQUEST "GET %s HTTP/1.0\r\nExtra-header: "
+#define GET_REQUEST "GET %s HTTP/1.0\r\nHost: %s\r\nConnection: keep-alive\r\nExtra-header: "
```

Comment:
How is changing the headers related to adding the output file feature? This seems like an unrelated change and deserves its own commit message with its own justification written in the commit message.

Reply:
If the request header is missing the host field, many http servers will refuse to return the resource you requested, causing the request to fail. So I added it.